flatnav/Index.h

```diff
@@ -326,6 +323,10 @@ template <typename dist_t, typename label_t> class Index {
               [](const dist_label_t &left, const dist_label_t &right) {
                 return left.first < right.first;
               });
+    if (results.size() > static_cast<size_t>(K)) {
```

Comment:
I think this might actually be a crucial change. Previously, we were taking the returned priority queue of candidates (which might be > K), then reducing it to a queue of size K, then returning a sorted list of this queue. This is not the order in which we should doing things since it's possible that one of the removed elements prior to sorting might actually be in the top K. This change makes sure that we first sort the computed priority queue before resizing it to a list of K values.